project/MimaExcludes.scala

```diff
@@ -118,79 +96,23 @@ object MimaExcludes {
     ProblemFilters.exclude[Problem]("org.apache.spark.sql.execution.*"),
     ProblemFilters.exclude[Problem]("org.apache.spark.sql.internal.*"),
     ProblemFilters.exclude[Problem]("org.apache.spark.sql.errors.*"),
+    // SPARK-40283: add jdbc as default excludes
+    ProblemFilters.exclude[Problem]("org.apache.spark.sql.jdbc.*"),
```

Comment:
We do have developer APIs in this package (e.g. `JdbcDialect`), but dialect implementations are private (e.g. `private object DB2Dialect extends JdbcDialect`). I have no idea why mima still checks these dialect implementations...

Reply:
I have add `org.apache.spark.sql.jdbc.*` into default excludes list

Reply:
We have developer APIs there so we can't simply exclude the entire package. If we can't figure out how to fix mima (will it help to add `@Private` for these private classes/objects?), then we have to add exclude rules one by one for each class in `org.apache.spark.sql.jdbc`

Reply:
ignore the class with `org.apache.spark.annotation.Private` ? The code should be feasible, what do you think about this @dongjoon-hyun

Reply:
@cloud-fan Are all `XXDialect extends JdbcDialect ` identified as `@Private`? Let me test it first

Reply:
they are `private object`...

Reply:
So we want to skip all `private object`?

Reply:
yea, mima is for checking binary compatibility and why do we care about private APIs?

Reply:
Yes, that's reasonable. Let me check why `GenerateMIMAIgnore` didn't ignore them

Reply:
@cloud-fan 

[ad543c0](https://github.com/apache/spark/pull/37741/commits/ad543c057bed529e36172e1e6e463b9c2c71369d) add `moduleSymbol.isPrivate` condition to `directlyPrivateSpark` check, then `private object` will ignore as default.

But we still need explicitly add `ProblemFilters.exclude[Problem]("org.apache.spark.sql.jdbc.JdbcDialect.*")` due to JdbcDialect is a `abstract class` now

Reply:
And `@DeveloperApi(for example JdbcDialect)` can not be ignore as default due to https://github.com/apache/spark/pull/11751